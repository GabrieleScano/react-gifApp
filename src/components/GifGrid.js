import React from 'react'
import PropTypes from 'prop-types'  

import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import { Spinner } from './Spinner'

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{category}</h3>

      {loading && <Spinner />}

      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

