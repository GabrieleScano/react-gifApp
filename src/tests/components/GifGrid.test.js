import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Test <GifGrid /> component', () => {
    const category = 'Test'
    test('should show the component', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
      });
    
    test('should show items when loaded', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        })


        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
    })
    
    
})
