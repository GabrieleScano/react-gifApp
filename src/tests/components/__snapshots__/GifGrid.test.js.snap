// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> component should show items when loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Test
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test <GifGrid /> component should show the component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Test
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
